staging/src/github.com/kubeedge/mapper-framework-Java/src/main/java/data/dbmethod/Tdengine.java

```diff
@@ -0,0 +1,129 @@
+
+import static data.DataConverter.convertToString;
+import static model.Const.defaultReportCycle;
+@Slf4j
+public class Tdengine {
+    public static void dataHandler(DeviceInstance.Twin twin, CustomizedClient client, VisitorConfig visitorConfig, DataModel dataModel, List<ScheduledFuture<?>> futures){
+        DataBaseConfig dataBaseConfig = null;
+        Connection dbClient;
+        try {
+            dataBaseConfig = newDataBaseClient(twin.getProperty().getPushMethod().getDbMethod().getDbConfig().getTdEngineClientConfig());
+        } catch (IOException e) {
+            log.error("new database client error: {}",e.getMessage(),e);
+        }
+        if (dataBaseConfig != null) {
+            dbClient = dataBaseConfig.initDbClient();
+        } else {
+            dbClient = null;
+        }
+        Duration reportCycle = Duration.ofSeconds(twin.getProperty().getReportCycle());
+        if(reportCycle.isZero()){
+            reportCycle = defaultReportCycle;
+        }
+        ScheduledExecutorService scheduler_db = Executors.newSingleThreadScheduledExecutor();
+        DataBaseConfig finalDataBaseConfig = dataBaseConfig;
+        ScheduledFuture<?> future_dbHandler = scheduler_db.scheduleAtFixedRate(()->{
+            if (Thread.currentThread().isInterrupted()){
+                if (finalDataBaseConfig != null) {
+                    try {
+                        finalDataBaseConfig.closeSession(dbClient);
+                    } catch (SQLException e) {
+                        log.error("Fail to close TDEngine connection with error: {}",e.getMessage(),e);
+                    }
+                }
+                return;
+            }
+            Object deviceData = client.getDeviceData(visitorConfig);
+            String sData = convertToString(deviceData);
+            dataModel.setValue(sData);
+            dataModel.setTimeStamp(Instant.now().toEpochMilli());
+            finalDataBaseConfig.addData(dataModel,dbClient);
+        },0, reportCycle.toSeconds(), TimeUnit.SECONDS);
+        futures.add(future_dbHandler);
+    }
+    public static DataBaseConfig newDataBaseClient(byte[] config) throws IOException {
+        ObjectMapper objectMapper = new ObjectMapper();
+        objectMapper.registerModule(new ProtobufModule());
+        DataBaseConfig dataBaseConfig = new DataBaseConfig();
+        dataBaseConfig.setTdClientConfig(objectMapper.readValue(config,TDEngineClientConfig.class));
+        return dataBaseConfig;
+    }
+    @Getter
+    @Setter
+    public static class TDEngineClientConfig {
+        @JsonProperty("addr")
+        private String addr;
+        @JsonProperty("dbName")
+        private String dbName;
+    }
+
+    @Getter @Setter
+    public static class DataBaseConfig{
+        @JsonProperty("config")
+        private TDEngineClientConfig tdClientConfig;
+        public Connection initDbClient(){
+            Connection dbClient = null;
+            String userName = System.getenv("USERNAME");
```

Comment:
USERNAME and PASSWORD are suggested to be made constants

Reply:
Do you mean to set default values for USERNAME and PASSWORD?
Since USERNAME and PASSWORD are more private information, we want users to enter this information themselves and add it to the local environment.

Reply:
Defined as a constant, and here we reference this constant, not a magic string